x-pack/plugins/alerts/server/health/task.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Logger, KibanaRequest } from 'kibana/server';
+import {
+  RunContext,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '../../../task_manager/server';
+import { AlertsClient } from '../alerts_client';
+import { AlertsConfig } from '../config';
+import { GetBasePathFunction, HealthStatus } from '../types';
+
+export const HEALTH_TASK_TYPE = 'alerting_health_check';
+
+export const HEALTH_TASK_ID = `Alerting-${HEALTH_TASK_TYPE}`;
+
+export function initializeAlertingHealth(
+  logger: Logger,
+  taskManager: TaskManagerSetupContract,
+  getBasePath: GetBasePathFunction,
+  getAlertsClientWithRequest: (
+    request: KibanaRequest
+  ) => Promise<{
+    getAlertsClient: () => AlertsClient;
+  }>
+) {
+  const request = getFakeKibanaRequest(getBasePath);
+  registerAlertingHealthCheckTask(logger, taskManager, getAlertsClientWithRequest(request));
+}
+
+export function scheduleAlertingHealthCheck(
+  logger: Logger,
+  config: Promise<AlertsConfig>,
+  taskManager?: TaskManagerStartContract
+) {
+  if (taskManager) {
+    scheduleTasks(logger, taskManager, config);
+  }
+}
+
+function getFakeKibanaRequest(getBasePath: GetBasePathFunction) {
+  const requestHeaders: Record<string, string> = {};
+  return ({
+    headers: requestHeaders,
+    getBasePath: () => getBasePath(),
+    path: '/',
+    route: { settings: {} },
+    url: {
+      href: '/',
+    },
+    raw: {
+      req: {
+        url: '/',
+      },
+    },
+  } as unknown) as KibanaRequest;
+}
+
+function registerAlertingHealthCheckTask(
+  logger: Logger,
+  taskManager: TaskManagerSetupContract,
+  alertsClient: Promise<{
+    getAlertsClient: () => AlertsClient;
+  }>
+) {
+  taskManager.registerTaskDefinitions({
+    [HEALTH_TASK_TYPE]: {
+      title: 'Alerting framework health check task',
+      createTaskRunner: healthCheckTaskRunner(logger, alertsClient),
+    },
+  });
+}
+
+async function scheduleTasks(
+  logger: Logger,
+  taskManager: TaskManagerStartContract,
+  config: Promise<AlertsConfig>
+) {
+  try {
+    const interval = (await config).healthCheck.interval;
+    await taskManager.ensureScheduled({
```

Comment:
I was anxious to see this running, so hacked an alert executor to throw an error, and then the config to run every minute :-).  Makes me wonder if we want to schedule an additional one-time call at startup (maybe 1 minute after startup), so we can get the latest data shortly after startup, without having to wait for the hourly task to run.  (could be another issue/PR)